<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.5.1-0-g916b5db  Build Version: 1.5.7152.31018
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="infergy">
  <link name="world"/>
  <link name="end_point"/>
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0 0 0.0510765329857644"
        rpy="0 0 0" />
      <mass
        value="79.3945245858017" />
      <inertia
        ixx="4.98393702962181"
        ixy="1.91973549448359E-32"
        ixz="-2.85928996573851E-21"
        iyy="4.98393702962181"
        iyz="-6.66349314846052E-18"
        izz="9.8186397110035" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name = "dummy" type="fixed">
  <parent
      link="world" />
  <child
      link="base_link" />
  </joint>
  <link
    name="link1">
    <inertial>
      <origin
        xyz="-4.25969318994524E-05 0.22482607118716 3.28927082688946E-05"
        rpy="0 0 0" />
      <mass
        value="42.4032183306581" />
      <inertia
        ixx="4.50890897419026"
        ixy="-0.000216435482029465"
        ixz="0.000222425633857992"
        iyy="1.38090817253925"
        iyz="0.000167128214450409"
        izz="4.50902659526784" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/link1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/link1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rev1"
    type="revolute">
    <origin
      xyz="0 0 0.1"
      rpy="1.5708 0 0" />
    <parent
      link="base_link" />
    <child
      link="link1" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="link2">
    <inertial>
      <origin
        xyz="0.00235685893503057 0.618985917366044 0.00359502003122458"
        rpy="0 0 0" />
      <mass
        value="22.461269494883" />
      <inertia
        ixx="2.01017285081223"
        ixy="0.000139271672549591"
        ixz="-0.00578769749890506"
        iyy="0.0900595301809686"
        iyz="0.000212437174391792"
        izz="2.00513897884739" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/link2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/link2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="prism2"
    type="prismatic">
    <origin
      xyz="0 0.2 0"
      rpy="0 0 0" />
    <parent
      link="link1" />
    <child
      link="link2" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="0.8"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="link3">
    <inertial>
      <origin
        xyz="0.548973051492322 -0.0184160496962614 0.00758118055783608"
        rpy="0 0 0" />
      <mass
        value="7.40354002036845" />
      <inertia
        ixx="0.0374912716131671"
        ixy="0.0293299662429915"
        ixz="-4.44089209850063E-16"
        iyy="0.782990054819416"
        iyz="5.55111512312578E-17"
        izz="0.754914246556592" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/link3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/link3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rev3"
    type="revolute">
    <origin
      xyz="0.087723 1.06 0.13381"
      rpy="-2.1511 0.011486 -1.5783" />
    <parent
      link="link2" />
    <child
      link="link3" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.14"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="link4">
    <inertial>
      <origin
        xyz="0.210787481970997 0.153346085635631 -0.000658297256135087"
        rpy="0 0 0" />
      <mass
        value="3.05467924135365" />
      <inertia
        ixx="0.00393838196136871"
        ixy="1.009392222584E-16"
        ixz="-5.39308630452942E-05"
        iyy="0.0581135578970605"
        iyz="3.49655200626575E-18"
        izz="0.0566698306494639" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/link4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://infergy/meshes/link4.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rev4"
    type="revolute">
    <origin
      xyz="0.90889 0.063407 0.0075812"
      rpy="3.1416 -1.4872 -0.027469" />
    <parent
      link="link3" />
    <child
      link="link4" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="0"
      velocity="0" />
  </joint>
  <joint name = "end_joint" type="fixed">
  <origin
      xyz="0.40889 0.163407 0.0075812"
      rpy="3.1416 -3.0572 3.11300"/>
  <parent
      link="link4" />
  <child
      link="end_point" />
  </joint>
</robot>